frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,553 @@
+            },
+            {
+              target: "getBillingDetailData",
+            },
+          ],
+        },
+        getBillingDetailData: {
+          on: {
+            FETCH_BILLING_DETAIL_SUCCES: {
+              target: "#billing.billingFormReady",
+              actions: "assignBillingDetail",
+            },
+            FETCH_BILLING_DETAIL_ERROR: {
+              target: "getBillingDetailDataError",
+              actions: "assignBillingDetailErrorMessage",
+            },
+          },
+        },
+        getBillingDetailDataError: {
+          on: {
+            REFTECH_BILLING_DETAIL: "getBillingDetailData",
+          },
+        },
+      },
+    },
+    billingFormReady: {
+      initial: "firstStep",
+      states: {
+        firstStep: {
+          on: {
+            NEXT_STEP: {
+              actions: "fillTheFirstForm",
+              target: "secondStep",
+            },
+          },
+        },
+        secondStep: {
+          on: {
+            PREV_STEP: "firstStep",
+            SUBMIT_BILLING: {
+              target: "#billing.submitBilling",
+              actions: "fillTheSecondForm",
+            },
+          },
+        },
+      },
+    },
+    submitBilling: {
+      on: {
+        SUBMIT_BILLING_SUCCES: {
+          target: "submitBillingOK",
+        },
+        SUBMIT_BILLING_ERROR: {
+          target: "submitBillingError",
+          actions: "assignSubmitBillingError",
+        },
+      },
+    },
+    submitBillingOK: {
+      on: {
+        BACK_TO_BILLING_SCREEN: {
+          target: "getBillingOK",
+        },
+      },
+    },
+    submitBillingError: {
+      on: {
+        BACK_TO_SECOND_STEP_FORM: "billingFormReady.secondStep",
+        REFETCH_SUBMIT_BILLING: "submitBilling",
+      },
+    },
+  },
+}).withConfig({
+  actions: {
+    assignBillingsData: assign({
+      billings: (ctx, event) =>
+        event.type === "FETCH_BILLING_SUCCES"
+          ? event.billingsData
+          : ctx.billings,
+    }),
+    assignBillingsErrorMessage: assign({
+      billingError: (ctx, event) =>
+        event.type === "FETCH_BILLING_ERROR"
+          ? event.billingsErrorMessage
+          : ctx.billingError,
+    }),
+    assignBillingDetail: assign({
+      billingDetail: (ctx, event) =>
+        event.type === "FETCH_BILLING_DETAIL_SUCCES"
+          ? event.billingDetailData
+          : ctx.billingDetail,
+    }),
+    assignBillingDetailErrorMessage: assign({
+      billingDetailError: (ctx, event) =>
+        event.type === "FETCH_BILLING_DETAIL_ERROR"
+          ? event.billingDetailErrorMessage
+          : ctx.billingDetailError,
+    }),
+    assignMembersData: assign({
+      members: (ctx, event) =>
+        event.type === "FETCH_MEMBER_SUCCESS" ? event.membersData : ctx.members,
+    }),
+    assigMembersError: assign({
+      membersError: (ctx, event) =>
+        event.type === "FETCH_MEMBER_ERROR"
+          ? event.membersErrorMessage
+          : ctx.membersError,
+    }),
```

Comment:
Aku ada ide menarik buat gabungin beberapa action "assign" ini jadi satu. Jadi kita bisa bikin satu action bernama `updateContext`, terus kita bisa update value contextnya berdasarkan state (bisa pake pattern match), contohnya kayak gini nih : 

https://codesandbox.io/s/competent-jepsen-hr3v43?file=/src/fetchMachine.ts

Coba cek baris 75, disitu ada actions `updateContext`

Wdyt ?

Reply:
Poin plusnya apa ? 
- Kita gak perlu mikirin nama2 dari action nya
- Kita bisa dengan mudah melihat relasi antara action dan perubahan context di satu tempat

Reply:
Wahh, interesting. Boleh boleh.

Aku coba kerjain dulu ya.

Thank you Ka

Reply:
Cusssss sikaaattt 🚀